lib/src/view/play/challenge_screen.dart

```diff
@@ -406,15 +407,47 @@ class _ChallengeBodyState extends ConsumerState<_ChallengeBody> {
                                   );
                                 }
                               : null
-                          : snapshot.connectionState == ConnectionState.waiting
-                              ? null
-                              // TODO handle correspondence time control
-                              : () async {
+                          : timeControl ==
+                                  ChallengeTimeControlType.correspondence
+                              ? () async {
+                                  final createGameService =
+                                      ref.read(createGameServiceProvider);
                                   showPlatformSnackbar(
                                     context,
-                                    'Correspondence time control is not supported yet',
+                                    'Sent challenge to ${widget.user.name}',
                                   );
-                                },
+                                  final response =
```

Comment:
You should use the `_pendingCreateGame` of the `FutureBuilder` to ensure the button cannot be clicked twice (which should be renamed to `_pendingCreateChallenge`). So:

```dart
_pendingCreateChallenge = createGameService.newChallgenge(...);

await _pendingCreateChallenge;

if (context.mounted) {

 ... do stuff
}
```


This file starts to be hard to read, and the indentation is horrible. Some widgets should be extracted, but don't worry about it I'll refactor it later after this PR is merged.